src/sources/syslog.rs

```diff
@@ -568,7 +567,7 @@ mod test {
           "#,
         )
         .unwrap();
-        assert!(config.mode.is_tcp());
+        assert!(matches!(config.mode, Mode::Tcp { .. }));
```

Comment:
If this is all it takes to avoid another `derive` procedural macro, then good riddance.

Reply:
Yeah, I stumbled backward into this realization and was very pleased.